sceptre_cdk_handler/cdk_builder.py

```diff
@@ -24,93 +27,44 @@ class CdkBuilder(ABC):
     """A base class for CDK builders to define the interface they all must meet."""
     STACK_LOGICAL_ID = 'CDKStack'
 
-    @abstractmethod
-    def build_template(
-        self,
-        stack_class: Type[SceptreCdkStack],
-        cdk_context: Optional[dict],
-        sceptre_user_data: Any
-    ) -> dict: ...
-
-
-class BootstrappedCdkBuilder(CdkBuilder):
-    """A CdkBuilder for stacks utilizing the CDK bootstrap stack for asset-related actions."""
     def __init__(
         self,
         logger: logging.Logger,
         connection_manager: ConnectionManager,
         *,
         subprocess_run=subprocess.run,
-        app_class=aws_cdk.App,
         environment_variables=os.environ
     ):
         self._logger = logger
         self._connection_manager = connection_manager
         self._subprocess_run = subprocess_run
-        self._app_class = app_class
         self._environment_variables = environment_variables
 
+    @abstractmethod
     def build_template(
         self,
-        stack_class: Type[SceptreCdkStack],
-        cdk_context: Optional[dict],
-        sceptre_user_data: Any
-    ) -> dict:
-        assembly = self._synthesize(stack_class, cdk_context, sceptre_user_data)
-        self._publish(assembly)
-        template = self._get_template(assembly)
-        return template
-
-    def _synthesize(
-        self,
-        stack_class: Type[SceptreCdkStack],
         cdk_context: Optional[dict],
         sceptre_user_data: Any
-    ) -> CloudAssembly:
-        self._logger.debug(f'CDK synthesizing stack class: {stack_class.__name__}')
-        self._logger.debug(f'CDK Context: {cdk_context}')
-        app = self._app_class(context=cdk_context)
-        stack_class(app, self.STACK_LOGICAL_ID, sceptre_user_data)
-        return app.synth()
-
-    def _publish(self, cloud_assembly: CloudAssembly):
-        asset_artifacts = self._get_assets_manifest(cloud_assembly)
-        if self._only_asset_is_template(asset_artifacts):
-            # Sceptre already has a mechanism to upload the template if configured. We don't
-            # need to deploy assets if the only asset is the template
-            self._logger.debug("Only asset is template; Skipping asset upload.")
-            return
+    ) -> dict: ...
```

Comment:
This method is now abstract, since there are different ways the template is built, depending on the kind of builder being employed.